static/merge.js

```diff
@@ -304,64 +273,291 @@
      * In addition each sublevel and his observations
      * are added to mergedHorizontalLevels dictionary.
      */
-    function getCells(headers, levelId, constructId, construct) {
+    function getCells(headers, constructId, construct) {
         var stars = [];
 
         [...Array(headers.length).keys()].map(i => {
-            initMergedLevel(levelId, i, headers);
-            stars[i] = removeColumns(headers, i, constructId, construct);
+            stars[i] = findCells(headers, i, constructId, construct);
         });
 
         return stars;
     }
 
+    function getTdsWhichEndClass(stars) {
+        return [...Array(stars[0].length).keys()].filter(i => $(stars[0][i]).hasClass('thicker'));
+    }
+
+    function restoreSublevelHeader(isLast, constructId, header) {
+        if (isLast) {
+            $($(`.star_chart_table-${constructId} tr:eq(1)`).first())
+                .find('th')
+                .eq(startIndex)
```

Comment:
You can reduce number of lines with `if` statement which sets `startIndex` first.